cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/log/LogHandlerTestRun.java

```diff
@@ -277,9 +321,18 @@ private void testPrev(String appId, String entityType, String entityId, String n
     }
   }
 
-  private void testPrevRunId(String appId, String entityType, String entityId, String namespace) throws Exception {
-    String nextNoFromUrl = String.format("apps/%s/%s/%s/runs/0/logs/prev?max=100", appId, entityType, entityId);
-    HttpResponse response = doGet(getVersionedAPIPath(nextNoFromUrl, namespace));
+  private String handleAdapterId(String prevUrl, @Nullable String adapteId, String joiner) {
```

Comment:
actually, is this necessary? It seems strange that you have to pass in a joiner/delimiter.. so someone could pass anything that they want.

It may be ok to just have this logic everywhere. Thoughts?